pkg/controllermanager/controller/shoot/shoot_conditions_control.go

```diff
@@ -141,3 +134,33 @@ func FilterSeedForShootConditions(obj, oldObj, _ client.Object, _ bool) bool {
 
 	return false
 }
+
+// MergeSeedConditions merges the conditions from the seed in the shoot avoiding duplications
+func MergeSeedConditions(seed *gardencorev1beta1.Seed, shoot *gardencorev1beta1.Shoot) []gardencorev1beta1.Condition {
+	seedConditions := []gardencorev1beta1.Condition{}
+	if seed != nil {
+		seedSystemComponentsCondition := gardencorev1beta1helper.GetOrInitCondition(seed.Status.Conditions, gardencorev1beta1.SeedSystemComponentsHealthy)
+		seedSystemComponentsConditionInShoot := gardencorev1beta1.Condition{
```

Comment:
`gardencorev1beta1helper.GetOrInitCondition(seed.Status.Conditions, gardencorev1beta1.SeedSystemComponentsHealthy)` returns a copy of the condition so you could simply change the type instead of creating a new condition.